### PR TITLE
Implement notifications

### PR DESCRIPTION
**Pull request by MichaelaTSH**

### CODE REVIEW

File: webproject/src/main/webapp/create-room.html [hunk 1]
```diff
@@ -27,10 +27,16 @@
     <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
     <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
     <script src="js/create-room.js"></script>
+    <script src="js/notification.js"></script>
     <script src="js/logout.js"></script>
 </head>
 
 <body data-spy="scroll" data-target=".site-navbar-target" data-offset="300" onload="appendUsernameToLogoutButton();">
+    <!-- The core Firebase JS SDK is always required and must be listed first -->
+    <script src="https://www.gstatic.com/firebasejs/7.15.5/firebase-app.js"></script>
```
Comment (kevinluvian) [inline webproject/src/main/webapp/create-room.html:36]: are there other places which doesnt have this?

**Review by kevinluvian — CHANGES_REQUESTED** [review]

File: webproject/src/main/java/com/google/sps/data/Room.java [hunk 2]
```diff
@@ -35,6 +36,7 @@ private Room (Builder builder) {
         this.category = builder.category;
         this.minPrice = builder.minPrice;
         this.deliveryFee = builder.deliveryFee;
+        this.timestamp = builder.timestamp;
```
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/data/Room.java:41]: does the PR need to have this file modified?

**Review by kevinluvian — CHANGES_REQUESTED** [review]

File: webproject/src/main/java/com/google/sps/data/Room.java (hunk 2 above)
Comment (MichaelaTSH) [reply webproject/src/main/java/com/google/sps/data/Room.java:41]: How I implemented the notification is that the user will be notified when a room is created after the webpage has been loaded, to ensure that the user is not notified for rooms that they are already aware of or that existed before they accessed that webpage. That is why I added the timestamp for the creation of a Room and left it in the PR relating to notifications. Should I move it to a separate PR?

Comment (kevinluvian) [reply webproject/src/main/java/com/google/sps/data/Room.java:41]: got it. sorry I missed it.

File: webproject/src/main/java/com/google/sps/servlets/RoomServlet.java [hunk 3]
```diff
@@ -35,6 +35,7 @@ public void doPost (HttpServletRequest request, HttpServletResponse response) th
                 .setMinPrice(Double.parseDouble(request.getParameter("minPrice")))
                 .setDeliveryFee(Double.parseDouble(request.getParameter("deliveryFee")))
                 .setCreator(userEmail)
+                .setTimestamp(System.currentTimeMillis())
```
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/servlets/RoomServlet.java]: use timestampUtil

**Review by kevinluvian — CHANGES_REQUESTED** [review]
add tests

**Review by kevinluvian — APPROVED** [review]

**Pull request merged**